src/CheckboxSelector/widget/checkboxselector.js

```diff
@@ -1,459 +1,418 @@
+		headers: '',
+		displayWidth: '',
+		displayAttrs: '',
+		currency: 'None',
+		useSeparators: '',
+		constraint: '',
+		limit: 0,
+		onchangemf: '',
+		listenchannel: '',
+		beginEmpty: false,
+		hasSelectAll: true,
+		unused: null,
+
+		//Caches
+		context: null,
+		_hasStarted: false,
+		assoc: null,
+		referredEntity: null,
+		rowsList: null,
+		ignoreChange: false,
+		isInactive: false,
+		isDisabled: false,
+		readOnlyBool: false,
+		validationDiv: null,
+		changeHandler: null,
+		changeHandler2: null,
+
+		readonly: false,
+
+
+		startup: function () {
+			if (this._hasStarted)
+				return;
+
+			this._hasStarted = true;
+			domConstruct.empty(this.domNode);
+			this.rowsList = {};
+			domClass.add(this.domNode, 'checkboxSelector_widget');
+		},
+
+		update: function (obj, callback) {
+			//Make sure the table is only rendered ONCE.
+			if (query(":first-child", this.domNode).length == 0) {
+				this.getAssocObjs(obj);
+			}
+			callback && callback();
+		},
+
+		postMixInProperties: function () {
+			this.inherited(arguments);
+			if (this.boxWidthUnit == "u0000") {
+				this.boxWidthUnit = "";
+			} else if (this.boxWidthUnit == "u0025") {
+				this.boxWidthUnit = "%";
+			}
+		},
+
+		getAssocObjs: function (context) {
+			this.context = context;
+			var contextGUID = context.getGuid();
+
+			var nameSplit = this.nameAttr.split("/");
+			this.assoc = nameSplit[0];
+			this.referredEntity = nameSplit[1];
+
+			this.readOnlyBool = context.isReadonlyAttr(this.assoc);
+
+			if (this.readOnlyBool == true || this.readonly == true || this.isDisabled == true)
+				this.isInactive = true;
+			else
+				this.isInactive = false;
+
+			this.unsubscribe(this.changeHandler);
+			this.unsubscribe(this.changeHandler2);
+			this.unsubscribe(this.validationHandler);
+			this.changeHandler = this.subscribe({ guid: contextGUID, attr: this.assoc, callback: lang.hitch(this, this.changeReceived) });
+			this.changeHandler2 = this.subscribe({ guid: contextGUID, callback: lang.hitch(this, this.objChangeReceived) });
+			this.validationHandler = this.subscribe({ guid: contextGUID, val: true, callback: lang.hitch(this, this.validationUpdate) });
+
+			var xpath = "//" + this.referredEntity + this.constraint.replace(/\[\%CurrentObject\%\]/gi, contextGUID);
+
+			mx.data.get({
+				'xpath': xpath,
+				'filter': {
+					'limit': this.limit + "",
+					'sort': [[this.sortAttr, this.sortOrder]]
+				},
+				'error': function () {
+					logger.error("Checkbox Selector Widget: Retrieve objects failed.");
+				},
+				'callback': lang.hitch(this, this.renderList)
+			});
+
+		},
+
+
+		renderList: function (objs) {
+			domConstruct.empty(this.domNode);
+			var selectedObjs = this.context.getReferences(this.assoc);
+
+			var surrDiv = domConstruct.create('div', { 'class': 'mendixDataGrid_tablePane' });
+			var table = domConstruct.create('table', { 'class': 'mendixDataGrid_gridTable table-bordered', 'cellPadding': '0px', 'cellSpacing': '0px' }, surrDiv);
```

Comment:
Do not swap order of `table` and `surrDiv`

Reply:
Changed